leader-election-impl/src/test/java/com/palantir/paxos/FileToSqlitePaxosStateLogIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class FileToSqlitePaxosStateLogIntegrationTest {
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private PaxosStateLog<PaxosValue> source;
+    private PaxosStateLog<PaxosValue> target;
+    private SqlitePaxosStateLogMigrationState migrationState;
+
+    @Before
+    public void setup() throws IOException {
+        source = new PaxosStateLogImpl<>(tempFolder.newFolder("source").getPath());
+        Supplier<Connection> targetConnSupplier = SqliteConnections
+                .createDefaultNamedSqliteDatabaseAtPath(tempFolder.newFolder("target").toPath());
+        target = SqlitePaxosStateLog.create(NAMESPACE, targetConnSupplier);
+        migrationState = SqlitePaxosStateLogMigrationState.create(NAMESPACE, targetConnSupplier);
+    }
+
+    @Test
+    public void emptyMigrationSucceeds() {
+        migrate();
+        assertThat(migrationState.hasAlreadyMigrated()).isTrue();
+    }
+
+    @Test
+    public void contiguousMigrationFromZeroSucceeds() {
+        migrateAndVerifyValuesForSequences(LongStream.rangeClosed(0, 100));
+    }
+
+    @Test
+    public void contiguousMigrationFromGreaterThanZeroSucceeds() {
+        migrateAndVerifyValuesForSequences(LongStream.rangeClosed(50, 130));
+    }
+
+    @Test
+    public void nonContiguousMigrationSucceeds() {
+        migrateAndVerifyValuesForSequences(LongStream.iterate(0, x -> x + 1_301).limit(100));
+    }
+
+    private void migrateAndVerifyValuesForSequences(LongStream sequences) {
+        List<PaxosRound<PaxosValue>> rounds = generateRounds(sequences);
+        List<PaxosValue> expectedValues = roundsToValues(rounds);
+
+        source.writeBatchOfRounds(rounds);
+
+        migrate();
+        List<PaxosValue> migratedValues = readMigratedValuesFor(expectedValues);
+
+        assertThat(migratedValues).isEqualTo(roundsToValues(rounds));
+        assertThat(migrationState.hasAlreadyMigrated()).isTrue();
+    }
+
+    private List<PaxosValue> roundsToValues(List<PaxosRound<PaxosValue>> rounds) {
+        return rounds.stream().map(PaxosRound::value).collect(Collectors.toList());
+    }
+
+    private void migrate() {
+        PaxosStateLogMigrator.migrate(ImmutableMigrationContext.<PaxosValue>builder()
+                .sourceLog(source)
+                .destinationLog(target)
+                .hydrator(PaxosValue.BYTES_HYDRATOR)
+                .migrationState(migrationState)
+                .build());
+    }
+
+    private List<PaxosValue> readMigratedValuesFor(List<PaxosValue> values) {
+        return values.stream()
+                .map(value -> PaxosValue.BYTES_HYDRATOR.hydrateFromBytes(readRoundUnchecked(target, value.seq)))
```

Comment:
not actionable: it's ok that this explodes if null is returned (since this is a test and we don't want nulls)